utils/fixed_decimal/src/decimal.rs

```diff
@@ -1941,7 +1941,13 @@ impl FixedDecimal {
                 }
 
                 let position = decimal.magnitude.wrapping_sub((sig - 1) as i16);
+                let old_magnitude = decimal.magnitude;
                 decimal.half_even(position);
+
+                // This means the significant digits has been increased by 1.
+                if decimal.magnitude > old_magnitude {
+                    decimal.truncate_right(position + 1);
```

Comment:
Suggestion (optional): you don't need to call `truncate_right`, which is more complicated. You can just do

```suggestion
                    decimal.lower_magnitude = cmp::min(0, position + 1);
```

Observe that the _only_ possible value after increasing magnitude is 1 followed by one or more zeros.

If you do this, please add a test case for 0.09 with significant digits 1 (it should end up being 0.1)